src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/controller/ReferralController.kt

```diff
@@ -83,4 +84,27 @@ constructor(
       ),
     )
   }
+
+  override fun getReferralSummariesByUsername(
+    @RequestParam(value = "page", defaultValue = "0") page: Int,
+    @RequestParam(value = "size", defaultValue = "10") size: Int,
+    @RequestParam(value = "status", required = false) status: List<String>?,
+    @RequestParam(value = "audience", required = false) audience: String?,
+    @RequestParam(value = "courseName", required = false) courseName: String?,
+  ): ResponseEntity<PaginatedReferralSummary> {
+    val pageable = PageRequest.of(page, size, Sort.by("referralId"))
+    val username: String = SecurityContextHolder.getContext().authentication?.name ?: "UNKNOWN_USER"
```

Comment:
Do we need `UNKNOWN_USER` here? If the username isn't present in the token, I wonder if it would be better to throw an error or prevent anything happening?

Reply:
We'd probably want to have a test for this case too.